### DISCUSSION TITLE
casadi won't run in python

### DISCUSSION THREAD

**Discussion by snDevX** [casadi/casadi#3803] [Q&A]
created a venv in python and run pip install casadi:

When I try to run an example I get the following error:

```
Exception of type: DYNAMIC_LIBRARY_FAILURE in file "/work/build/external_projects/src/ipopt-external/src/Common/IpLibraryLoader.cpp" at line 72:
 Exception message: libhsl.so: cannot open shared object file: No such file or directory
```

what should I do?

**Comment by tmmsartor** [comment]
It seems you are trying to use a solver from HSL library, like `ma27` or `ma57` or similar, and casadi is not able to find the HSL library in your system. If `libhsl.so` is installed in a not standard path you can add it running the following before running your casadi code:
``` export LD_LIBRARY_PATH="$LD_LIBRARY_PATH:/path/to/hsl_folder"```
Also you can try to run prepending `LD_DEBUG=libs` as suggested in #3291, to get more debug information.
